### DISCUSSION TITLE
What is the difference between INCLUDE and SINGLE_INCLUDE?

### DISCUSSION THREAD

**Discussion by petrasvestartas** [nlohmann/json#4298] [General]
Hi,

I am new to this repository.

I see it is not header only library as stated in the documentations. I need to download includes.

Could someone describe how and which includes I must reference in my cmake? The reason to ask is because there are **simple_include** and **include** folder.

I am referring to this:
<img width="527" alt="image" src="https://github.com/nlohmann/json/assets/18013985/e975cd0b-adfe-4862-b23b-05eb7670dfe7">

and this:
<img width="117" alt="image" src="https://github.com/nlohmann/json/assets/18013985/362bb5e5-56de-4360-b8c8-b128534aba46">

**Comment by gregmarr** [comment]
>I see it is not header only library as stated in the documentations. I need to download includes.

Just because there are multiple headers, that doesn't make it not header-only.  There is also a single stand-alone header in the `single_include` directory, along with a forward declaration header.

You can use `json.hpp` as a stand-alone header, or you can use the CMake support as in https://github.com/nlohmann/json?tab=readme-ov-file#cmake

**Reply by petrasvestartas** [reply]
What @gregmarr would be the difference between the single include and the 3rd party dependencies:
<img width="102" alt="image" src="https://github.com/nlohmann/json/assets/18013985/f5a6fd3c-18cf-4e13-9974-753fd0ed21a9">

I need writing and reading json files. I am wondering what would be missing if I include only the single header?

**Reply by gregmarr** [reply]
> I am wondering what would be missing if I include only the single header?

Nothing.  It's the contents of the `include` directory combined into a single file.
